featuretools/tests/synthesis/test_dfs_method.py

```diff
@@ -323,6 +309,135 @@ def test_accepts_pd_dateoffset_training_window(datetime_es):
     assert (feature_matrix.index == feature_matrix_2.index).all()
 
 
+def test_warns_with_unused_primitives(es):
+    trans_primitives = ['num_characters', 'num_words', 'add_numeric']
+    agg_primitives = [Max, 'min']
+
+    warning_text = "Some specified primitives were not used during DFS. " + \
+        "trans_primitives: ['add_numeric'] agg_primitives: ['max', 'min']. " + \
+        "This may be caused by a using a value of max_depth that is too small, not setting interesting values, " + \
+        "or it may indicate no compatible variable types for the primitive were found in the data."
+
+    with pytest.warns(UserWarning) as record:
+        dfs(entityset=es,
+            target_entity='customers',
+            trans_primitives=trans_primitives,
+            agg_primitives=agg_primitives,
+            max_depth=1)
+
+    assert record[0].message.args[0] == warning_text
+
+    # Should not raise a warning
+    with pytest.warns(None) as record:
+        dfs(entityset=es,
+            target_entity='customers',
+            trans_primitives=trans_primitives,
+            agg_primitives=agg_primitives,
+            max_depth=2)
+
+    assert not record
+
+
+def test_warns_with_unused_where_primitives(es):
+    warning_text = "Some specified primitives were not used during DFS. " + \
+        "where_primitives: ['count', 'sum']. " + \
+        "This may be caused by a using a value of max_depth that is too small, not setting interesting values, " + \
+        "or it may indicate no compatible variable types for the primitive were found in the data."
+
+    with pytest.warns(UserWarning) as record:
+        dfs(entityset=es,
+            target_entity='customers',
+            agg_primitives=['count'],
+            where_primitives=['sum', 'count'],
+            max_depth=1)
+
+    assert record[0].message.args[0] == warning_text
+
+
+def test_warns_with_unused_groupby_primitives(pd_es):
+    warning_text = "Some specified primitives were not used during DFS. " + \
+        "groupby_trans_primitives: ['cum_sum']. " + \
+        "This may be caused by a using a value of max_depth that is too small, not setting interesting values, " + \
+        "or it may indicate no compatible variable types for the primitive were found in the data."
+
+    with pytest.warns(UserWarning) as record:
+        dfs(entityset=pd_es,
+            target_entity='sessions',
+            groupby_trans_primitives=['cum_sum'],
+            max_depth=1)
+
+    assert record[0].message.args[0] == warning_text
+
+    # Should not raise a warning
+    with pytest.warns(None) as record:
+        dfs(entityset=pd_es,
+            target_entity='customers',
+            groupby_trans_primitives=['cum_sum'],
+            max_depth=1)
+
+    assert not record
+
+
+def test_warns_with_unused_custom_primitives(pd_es):
```

Comment:
does this test not work with the Dask entityset?

Reply:
As of now, `make_trans_primitive` and `make_agg_primitive` can be used to create a Dask compatible primitive. We could certainly define a custom primitive that would work without those functions, but I wasn't sure it would add much value beyond testing with pandas.